domain/schema/model/sql/0024-relation.sql

```diff
@@ -175,8 +195,8 @@ FROM relation AS r
 JOIN relation_endpoint AS re ON r.uuid = re.relation_uuid
 JOIN application_endpoint AS ae ON re.endpoint_uuid = ae.uuid
 JOIN charm_relation AS cr ON ae.charm_relation_uuid = cr.uuid
-JOIN charm_relation_role AS crr ON cr.role_id = crr.id
-JOIN charm_relation_scope AS crs ON cr.scope_id = crs.id
+LEFT JOIN charm_relation_role AS crr ON cr.role_id = crr.id -- can be null
+LEFT JOIN charm_relation_scope AS crs ON cr.scope_id = crs.id -- can be null
```

Comment:
todo: A charm relation must have a role, it cannot exist without one. Please update the charm schema in a separate PR, rather than making this change.

https://github.com/juju/juju/blob/6d9f0fea0959a79d9ab3d66a59461feeb56e1bbc/internal/charm/meta.go#L790-L814

I don't see how a charm's relation can relate without a scope. I believe this should be updated in the charm schema as well. Chat with Simon and maybe update the schema or pass to him to update.

Same comments apply to the new `v_application_endpoint`.